local/compose/create.go

```diff
@@ -129,6 +135,30 @@ func prepareNetworks(project *types.Project) {
 	}
 }
 
+func prepareNetworkFrom(p *types.Project) error {
+outLoop:
+	for i := range p.Services {
+		dependency := getNetworkFrom(p.Services[i].NetworkMode)
+		if dependency != "" {
+			if p.Services[i].DependsOn == nil {
+				p.Services[i].DependsOn = make(types.DependsOnConfig)
+			}
+
+			for _, service := range p.Services {
```

Comment:
can just usse `p.GetService(dependency)`

Reply:
The idea here is to change the project's collection. If I use `Project::GetService` I will set the dependency into a copy of the service, not the one attached to the project